front/src/components/Profile/Profile.tsx

```diff
@@ -0,0 +1,622 @@
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(email);
+  };
+
+  const validateProfile = () => {
+    let newErrors: Errors = {};
+
+    if (!editedData.user_name?.trim()) {
+      newErrors.user_name = "Username is required";
+    }
+    if (!editedData.display_name?.trim()) {
+      newErrors.display_name = "Display Name is required";
+    }
+    if (!editedData.email?.trim()) {
+      newErrors.Email = "Email is required";
+    } else if (!validateEmail(editedData.email)) {
+      newErrors.Email = "Invalid email address";
+    }
+    setErrors(newErrors);
+
+    return Object.keys(newErrors).length === 0;
+  };
+
+  const editUser = async (id: number, updatedData: RowData) => {
+    if (validateProfile()) {
+      try {
+        const response = await UserApi.update(id, updatedData);
+        if (response) {
+          if (response.statusCode == 409) {
+          } else if (response.statusCode == 200) {
+            setRows(response.data);
+            const updatedUserDetails = { ...userDetails, ...updatedData };
+            setUserDetails(updatedUserDetails);
+            setAlertShow(response.message);
+          }
+        }
+      } catch (error: any) {
+        var response = error.response.data;
+        if (
+          error.response &&
+          error.response.data &&
+          error.response.data.statusCode === 422
+        ) {
+          console.error(error);
+        }
+      }
+    }
+  };
+
+  const validate = () => {
+    let newErrors: Errors = {};
+    const passwordRegex =
+      /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[~`!@#$%^&*()-_+={}[\]|\\:;"'<>,.?/]).{8,}$/;
+
+    if (!currentPassword.trim()) {
+      newErrors.currentPassword = "Current Password is required";
+    }
+
+    if (!newPassword.trim()) {
+      newErrors.newPassword = "Password is required";
+    } else if (!passwordRegex.test(newPassword)) {
+      newErrors.newPassword =
+        "Password must contain at least 8 characters, including 1 uppercase letter, 1 lowercase letter, 1 number, and 1 special character";
+    }
+
+    if (!confirmPassword.trim()) {
+      newErrors.confirmPassword = "Confirm Password is required";
+    } else if (newPassword !== confirmPassword.trim()) {
+      newErrors.confirmPassword = "Passwords do not match";
+      if (!areCharactersMatching(newPassword, confirmPassword)) {
+        newErrors.confirmPassword =
+          "Password should match with the confirm password";
+      }
+    }
+    setErrors(newErrors);
+    return Object.keys(newErrors).length === 0;
+  };
+
+  const areCharactersMatching = (
+    newPassword: string,
+    confirmPassword: string
+  ): boolean => {
+    if (newPassword.length !== confirmPassword.length) {
+      return false;
+    }
+    for (let i = 0; i < newPassword.length; i++) {
+      if (newPassword[i] !== confirmPassword[i]) {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  const editPassword = async (
```

Comment:
Give a relevant name for the function

Reply:
relevant name for function has been changed